### PR TITLE
internal: add PrintPeekPoke annotation for debugging

### PR DESCRIPTION
**Pull request by ekiwi**
Some convenient functionality that I recently used to track down a bug. Might come in handy to others and it is part of the `internal` package, so it won't increase our public API surface.

### CODE REVIEW

**Review by chick — APPROVED** [review]
Looks clean, and useful.

File: src/main/scala/chiseltest/simulator/DebugPrintWrapper.scala [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  }
+  override def peekMemory(signal: String, index: Long) = {
+    val res = inner.peekMemory(signal, index)
+    println(s"$signal[$index] -> $res")
+    res
+  }
+
+  override def pokeMemory(signal: String, index: Long, value: BigInt): Unit = {
+    println(s"$signal[$index] <- $value")
+    inner.pokeMemory(signal, index, value)
```
Comment (ducky64) [inline src/main/scala/chiseltest/simulator/DebugPrintWrapper.scala:33]: Do you want to be consistent in whether the print happens before or after the pokes / other actions?

File: src/main/scala/chiseltest/simulator/DebugPrintWrapper.scala [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    println("finish()")
+  }
+  override def resetCoverage(): Unit = {
+    inner.resetCoverage()
+    println("resetCoverage()")
+  }
+  override def getCoverage() = {
+    val res = inner.getCoverage()
+    println("getCoverage()")
+    // println(res.mkString("\n"))
```
Comment (ducky64) [inline src/main/scala/chiseltest/simulator/DebugPrintWrapper.scala:46]: dead code

**Review by ducky64 — APPROVED** [review]
printfs were also useful for me in development, and this is a pretty architecturally clean way to do it!

Some nits in the comments though.

File: src/main/scala/chiseltest/simulator/DebugPrintWrapper.scala (hunk 1 above)
Comment (ekiwi) [reply src/main/scala/chiseltest/simulator/DebugPrintWrapper.scala:33]: good catch

**Pull request merged**